src/core/infra/repositories/ApiRepository.ts

```diff
@@ -35,7 +26,28 @@ export abstract class ApiRepository {
       });
   }
 
+  private buildRequestConfig(authRequired: boolean): AxiosRequestConfig {
+    let requestConfig: AxiosRequestConfig = {
+      headers: { 'Content-Type': 'application/json' },
+    };
+    if (!authRequired) {
+      return requestConfig;
+    }
+    switch (ApiConfig.dataverseApiAuthMechanism) {
+      case DataverseApiAuthMechanism.SESSION_COOKIE:
+        /*
+          We set { withCredentials: true } to send the JSESSIONID cookie in the requests for API authentication. 
+          This is required, along with the session auth feature flag enabled in the backend, to be able to authenticate using the JSESSIONID cookie.
+          Auth mechanisms like this are configurable to set the one that fits the particular use case of js-dataverse. (For the SPA MVP, it is the session cookie API auth).
+        */
+        requestConfig.withCredentials = true;
+      case DataverseApiAuthMechanism.API_KEY:
+        requestConfig.headers['X-Dataverse-Key'] = ApiConfig.dataverseApiKey;
+    }
+    return requestConfig;
```

Comment:
No breaks after each switch case? Is that intentional? If we choose the cookie auth then it is also sending the API KEY